service/history/workflow/metrics.go

```diff
@@ -77,19 +77,26 @@ func emitMutableStateStatus(
 func emitWorkflowCompletionStats(
 	metricsHandler metrics.Handler,
 	namespace namespace.Name,
-	namespaceState string,
-	taskQueue string,
-	workflowTypeName string,
-	status enumspb.WorkflowExecutionStatus,
+	completion completionMetric,
 	config *configs.Config,
 ) {
-	handler := GetPerTaskQueueFamilyScope(metricsHandler, namespace, taskQueue, config,
+	handler := GetPerTaskQueueFamilyScope(metricsHandler, namespace, completion.taskQueue, config,
 		metrics.OperationTag(metrics.WorkflowCompletionStatsScope),
-		metrics.NamespaceStateTag(namespaceState),
-		metrics.WorkflowTypeTag(workflowTypeName),
+		metrics.NamespaceStateTag(completion.namespaceState),
+		metrics.WorkflowTypeTag(completion.workflowTypeName),
 	)
 
-	switch status {
+	if completion.state == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED &&
+		completion.executionTime != nil && completion.closeTime != nil {
+
+		executionTime := completion.executionTime.AsTime()
+		closeTime := completion.closeTime.AsTime()
+		if closeTime.After(executionTime) {
+			metrics.WorkflowDuration.With(handler).Record(closeTime.Sub(executionTime))
```

Comment:
just double checking, it's intentional to use execution time instead of start time here?